dynamodb/client.go

```diff
@@ -0,0 +1,99 @@
+package dynamodb
+
+import (
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/session"
+	"github.com/aws/aws-sdk-go/service/dynamodb"
+	"github.com/aws/aws-sdk-go/service/dynamodb/dynamodbattribute"
+	"github.com/aws/aws-sdk-go/service/dynamodb/dynamodbiface"
+)
+
+const (
+	TableHashKey = "conceptId"
+)
+
+type ConcordancesModel struct {
+	UUID         string   `json:"conceptId"`
+	ConcordedIds []string `json:"concordedIds"`
+}
+
+type Client interface {
+	Read(uuid string) (ConcordancesModel, error)
+	Write(m ConcordancesModel) (ConcordancesModel, error)
+	Delete(uuid string) (ConcordancesModel, error)
+}
+
+type DynamoDBClient struct {
+	dynamoDbTable string
+	awsRegion     string
+	ddb           dynamodbiface.DynamoDBAPI
+}
+
+func NewDynamoDBClient(dynamoDbTable string, awsRegion string) Client {
+	sess := session.Must(session.NewSession(&aws.Config{Region: aws.String(awsRegion)}))
+	ddb := dynamodb.New(sess)
+	c := DynamoDBClient{dynamoDbTable: dynamoDbTable, awsRegion: awsRegion, ddb: ddb}
+	return &c
+}
+
+func (s *DynamoDBClient) Read(uuid string) (ConcordancesModel, error) {
+	m := ConcordancesModel{}
+	input := &dynamodb.GetItemInput{}
+	input.SetTableName(s.dynamoDbTable)
+	k, err := dynamodbattribute.Marshal(uuid)
+
+	if err != nil {
+		return m, err
+	}
+	input.SetKey(map[string]*dynamodb.AttributeValue{"conceptId": k})
+	output, err := s.ddb.GetItem(input)
+	if err != nil {
+		return m, err
+	}
+	err = dynamodbattribute.UnmarshalMap(output.Item, &m)
```

Comment:
I think a check that output.Item != nil is needed here, in case no results are returned.  From the docs: `If there is no matching item, GetItem does not return any data and there will be no Item element in the response.`